src/workspaces/useWorkspaces.models.ts

```diff
@@ -0,0 +1,16 @@
+import { FieldsArg } from 'src/libs/ajax/workspaces/providers/WorkspaceProvider';
+import { WorkspaceWrapper } from 'src/libs/workspace-utils';
+
+/**
+ * the expected hook return for useWorkpaces hook usage.
+ */
+export interface UseWorkspacesStateResult {
```

Comment:
Some things (these types, Environments' prop) are named "UseWorkspacesState", whereas the hook is named "useWorkspaces". Can we make names more consistent by removing the "State"?

```suggestion
export interface UseWorkspacesResult {
```

Reply:
to be honest, I am inclined to rename the hook to useWorkspacesState since it encompases the current state (cached, loading-newer-data, etc).  We've touched on this naming challenge a few times... we shorthand to useWorkspaces (useObjectX) but there would be better clarity of we named hooks as:
- useObjectXData - just a data requestor
- useObjectXState - more state complexities with ability to update/refresh, etc.

thoughts?

Reply:
In my opinion, "data" and "state" are so generic that those names don't convey any meaningful distinction to me. Also, generally "use thing data" seems redundant with just "use thing".

And when we're talking about injecting these into components, like with Environments, I think we'd want the component's expected interface to be more generic. Like the component shouldn't care if there's a store involved. All it needs is something that returns some workspaces and their status.

Reply:
fair enough, and I've had my brain in the same space as well... suggesting useFocusedWorkspaces or something like that to indicate it's the workspaces the app is focusing on.... something to indicate it is more then just a data call wrapper... but as you said, more named towards the consumer's usage.

Reply:
but... as we consider code reuse more and more... assuming those added app-state mechanics is probably not good either... which lands me to your suggestion...

useWorkspaces that returns UseWorkspacesResult...

LOL, See, I go there... thanks for your patience :)

Reply:
renamed